### PR TITLE
BBoxTransform add float16

### PR DESCRIPTION
**Pull request by omromano**
Summary: 
- Add support for float16 BBoxTransform and corresponding unittest.
- Change RoiBatchSplits to always be FP32

### CODE REVIEW

File: lib/Backends/Interpreter/tests/InterpreterOperatorTest.cpp [hunk 1]
```diff
@@ -31,4 +31,5 @@ std::set<std::string> glow::backendTestBlacklist = {
     "EmbeddingBagByteRowwiseOffsets_Float_End_Offset_Partial/0",
     "EmbeddingBagByteRowwiseOffsets_Float16_AccumFloat_End_Offset_Partial/0",
     "EmbeddingBagByteRowwiseOffsets_Float16_AccumFloat16_End_Offset_Partial/0",
+    "BBoxTransform_Float16/0",
```
Comment (jfix71) [inline lib/Backends/Interpreter/tests/InterpreterOperatorTest.cpp:34]: Why is this necessary? The interpreter should support this, no?

File: lib/Graph/Graph.cpp [hunk 2]
```diff
@@ -4801,8 +4801,9 @@ BBoxTransformNode *Function::createBBoxTransform(
 
   auto boxOutTy = getParent()->uniqueTypeWithNewShape(
       rois.getType(), {deltasDims[0], deltasDims[1]});
+  // Forcing roiBatchSplitsTy to always be Float.
   auto roiBatchSplitsTy =
-      getParent()->uniqueType(rois.getElementType(), {imInfoDims[0]});
+      getParent()->uniqueType(ElemKind::FloatTy, {imInfoDims[0]});
```
Comment (jfix71) [inline lib/Graph/Graph.cpp:4806]: @mortzur Can you comment on this change?

File: tests/unittests/OperatorTest.cpp [hunk 3]
```diff
@@ -1555,35 +1557,61 @@ TEST_P(OperatorTest, BBoxTransform) {
...
       0.0000,   344.4728, 413.0000, 347.5388, 337.9926, 114.3067, 413.0000,
       173.1735, 0.0000,   0.0000,   0.0000,   83.6907};
+
+  auto *ROIS = mod.createPlaceholder(ElemTy, roisDims, "rois", false);
+  bindings.allocate(ROIS)->getHandle<DataType>() = rois;
+
+  auto *DELTAS = mod.createPlaceholder(ElemTy, deltasDims, "deltas", false);
+  bindings.allocate(DELTAS)->getHandle<DataType>() = deltas;
+
+  auto *IMINFO = mod.createPlaceholder(ElemTy, imInfoDims, "imInfo", false);
```
Comment (jfix71) [inline tests/unittests/OperatorTest.cpp:1580]: nit: We tend to use lowercase for non-acronym variables.

File: lib/Backends/Interpreter/tests/InterpreterOperatorTest.cpp (hunk 1 above)
Comment (omromano) [reply lib/Backends/Interpreter/tests/InterpreterOperatorTest.cpp:34]: When I tested it failed - I think it is related to the next change you commented on:
getParent()->uniqueType(ElemKind::FloatTy, {imInfoDims[0]});

File: lib/Graph/Graph.cpp (hunk 2 above)
Comment (mortzur) [reply lib/Graph/Graph.cpp:4819]: roiBatchSplits is in fact an integer but it is consumed as float by other ops (i.e. caffe2 NMS). 
This raises a problem that I haven't thought of - how to not convert this specific output to FP16 in case that FP16 global conversion is required.
